### PR TITLE
queryChannels fix - Removing watch if connectionId is absent

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -892,11 +892,6 @@ export class StreamChat {
 
 		// Make sure we wait for the connect promise if there is a pending one
 		await this.wsPromise;
-
-		if (!this._hasConnectionID()) {
-			defaultOptions.watch = false;
```
Comment (vishalnarkhede) [inline src/client.js:897]: Customers are  getting plenty of query channels related issue. In our logs, it seems like there are lot of queryChannel calls being made with empty connectionId.  Error for this case on js-client look something  like this:

```js
QueryChannels failed with error: "Watch or Presence require you to specify a user id"
```

Error is a bit misleading. But as I said, real reason is absence of connectionId in request. 

If there is no connectionId, which mean `setUser()` didn't succeed, then we used to remove watch from defaultOptions. But if there is a `watch: true` in options (provided in parameters), then final payload will have `watch: true` as well,  resulting in api failure ... which in turn fails the application itself:

So instead we should simply remove the watch from final payload itself and continue with queryChannels call. We already have reconnection logic in place, once reconnection succeeds, queryChannels will run again with `watch: true`

File: src/client.js [hunk 2]
```diff
@@ -906,6 +901,10 @@ export class StreamChat {
 			...options,
 		};
 
+		if (!this._hasConnectionID()) {
+			payload.watch = false;
```
Comment (thesyncim) [inline src/client.js:905]: we also need to handle
payload.presence = false

Comment (thesyncim) [reply src/client.js:905]: presence also requires connection_id

Comment (vishalnarkhede) [reply src/client.js:905]: Done!

**Pull request closed without merging**